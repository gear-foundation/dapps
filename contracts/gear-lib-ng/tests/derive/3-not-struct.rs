use gear_lib::StorageProvider;

#[derive(Clone, Copy)]
struct Storage;

#[derive(StorageProvider)]
enum ContractEnum {
    A {
        #[storage_field]
        storage: Storage,
    },
    B,
}

#[derive(StorageProvider)]
union ContractUnion {
    #[storage_field]
    storage: Storage,
}

fn main() {}
