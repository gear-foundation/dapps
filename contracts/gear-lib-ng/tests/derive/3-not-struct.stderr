error: `StorageProvider` deriving is supported only with structs
 --> tests/derive/3-not-struct.rs:7:1
  |
7 | enum ContractEnum {
  | ^^^^

error: `StorageProvider` deriving is supported only with structs
  --> tests/derive/3-not-struct.rs:16:1
   |
16 | union ContractUnion {
   | ^^^^^
